phpstan.neon

```diff
@@ -10,6 +10,7 @@ parameters:
         - tests
     excludePaths:
         - tests/Fixtures
+    treatPhpDocTypesAsCertain: false
```

Comment:
I was actually wondering about this and I'm not convinced.
was this option introduced only at 1.10? what was the default behaviour on 1.9?

Reply:
It is much older than 1.9.

I wasn't convinced to use it after reading the description in [config](https://phpstan.org/config-reference#treatphpdoctypesascertain), but [this](https://github.com/phpstan/phpstan/issues/3005#issuecomment-590740468) comment explains that we can have it:

> `treatPhpDocTypesAsCertain` is for cases when you have precise phpDoc but you also have to check for other types because the type is only in phpDoc and the caller (which can be outside of the analyzed scope) doesn't adhere to the phpDoc. So you have no other option than make the analyser a little bit "dumber".